### ISSUE TITLE
copy file failed: Failed to execute 'setRequestHeader' on 'XMLHttpRequest': Value is not a valid ByteString

### ISSUE THREAD

**Issue by bigthing33** [perry-mitchell/webdav-client#126]
I call copyFile method successfuly  **but**
when basename is chinese like “新建文件夹”,
throw error ### Failed to execute 'setRequestHeader' on 'XMLHttpRequest': Value is not a valid ByteString

- The problem like: https://github.com/axios/axios/issues/1945

**Comment by bigthing33** [comment]
I workaround with "encodeURIComponent".
- encodeURIComponent(targetBasename) 
 like
```
await client.copyFile("/photos/新建文件夹/", "/%E6%96%B0%E5%BB%BA%E6%96%87%E4%BB%B6%E5%A4%B9/");
```

**Comment by perry-mitchell** [comment]
Thanks @bigthing33 - This is definitely a bug. We should be encoding that value.

**Comment by perry-mitchell** [comment]
Hmm, I seem to get 404 errors when encoding.. Seems there's some issue.

**Comment by perry-mitchell** [comment]
Released in 2.2.1.

**Comment by bigthing33** [comment]
The bug has been fixed, thank you!
